pkg/autodetect/main.go

```diff
@@ -100,12 +100,33 @@ func (b *Background) autoDetectCapabilities() {
 
 		b.detectElasticsearch(ctx, apiList)
 		b.detectKafka(ctx, apiList)
+		b.detectCronjobsVersion(ctx)
 	}
 
 	b.detectClusterRoles(ctx)
 	b.cleanDeployments(ctx)
 }
 
+func (b *Background) detectCronjobsVersion(ctx context.Context) {
+	apiGroupVersions := []string{"batch/v1", "batch/v1beta1"}
+	for _, apiGroupVersion := range apiGroupVersions {
+		groupAPIList, err := b.dcl.ServerResourcesForGroupVersion(apiGroupVersion)
+		if err != nil {
+			log.Errorf("Error getting %s api list: %v", apiGroupVersion, err)
```

Comment:
Is there a way to get a logger form context?

iam asking because of this https://github.com/jaegertracing/jaeger-operator/issues/1817

Reply:
Sorry, there might be, but I don't know how to do that.

Reply:
I think its something like
```golang
"sigs.k8s.io/controller-runtime/pkg/log"
...
logger := log.FromContext(ctx)
```

Reply:
Sure.  Can we do that as part of #1817 though?